bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java

```diff
@@ -164,17 +229,67 @@ public CloudBigtableScanConfiguration build() {
           scan = new Scan();
         }
         ReadRowsRequest.Builder builder = Adapters.SCAN_ADAPTER.adapt(scan, readHooks);
-        request = readHooks.applyPreSendHook(builder.build());
+        request =  StaticValueProvider.of(readHooks.applyPreSendHook(builder.build()));
       }
       return new CloudBigtableScanConfiguration(projectId, instanceId, tableId,
           request, additionalConfiguration);
     }
   }
 
-  // 'request' needs to be a runtime parameter because it depends on runtime parameters: project ID
-  // and instance ID.
   private final ValueProvider<ReadRowsRequest> request;
 
+  /**
+   * Provides an updated request by setting the table name in the existing request if the table name
+   * wasn't set.
```

Comment:
I think this is backwards, shouldn't the more specific with{ProjectId, InstanceId, etc} override the more general request?

Reply:
Nevermind, as discussed offline, this is the current behavior